src/ExceptionMechanism.php

```diff
@@ -26,18 +26,28 @@ final class ExceptionMechanism
      */
     private $handled;
 
+    /**
+     * @var array|null Arbitrary extra data that might help the user understand
```

Comment:
If we accept `array<string, mixed>` in the constructor, this variable should be typehinted the same